### PR TITLE
Material You design changes and widgets update

### PR DESCRIPTION
**Pull request by TipzRickyCheung**
I have took some time to modify the app so that it no longer uses MaterialComponents styling. However, some more changes may be needed. Please edit as you see fit.

For the widgets change, I've added a wide variant of the dual pods widget, which gets activated when the widget is larger than 3 columns.

### CODE REVIEW

**Comment by TipzRickyCheung** [comment]
# Screenshots for the changes

**Comment by TipzRickyCheung** [comment]
I completely forgot about the notification bar in light mode. To get the icons to show correctly, you would need:
```
<item name="android:windowLightStatusBar" tools:targetApi="m">?attr/isLightTheme</item>
```

File: app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt [hunk 1]
```diff
@@ -95,6 +96,14 @@ class WidgetProvider : AppWidgetProvider() {
         }
     }
 
+    private fun getCellsForSize(size: Int): Int {
+        var n = 2
+        while (70 * n - 30 < size) {
```
Comment (d4rken) [inline app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt:101]: Can you explain the algorithm here? Where does the 70 and 30 values come from?

File: app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt [hunk 2]
```diff
@@ -169,7 +188,8 @@ class WidgetProvider : AppWidgetProvider() {
     private fun createDualPodLayout(
         context: Context,
         podDevice: DualPodDevice,
-    ): RemoteViews = RemoteViews(context.packageName, R.layout.widget_pod_dual_layout).apply {
+        layout: Int
```
Comment (d4rken) [inline app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt:191]: Could be annotated with `@LayoutRes`

**Review by d4rken — COMMENTED** [review]
Thanks! Looks good :heart: 

Can you take some screenshots in light-mode?

**Comment by TipzRickyCheung** [comment]
> Thanks! Looks good ❤️
> 
> Can you take some screenshots in light-mode?

Here you go:

**Comment by TipzRickyCheung** [comment]
@d4rken

Slight design change:
Do you prefer if the wide weights looked like this?
Screenshot_20240626-173050

File: app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt (hunk 2 above)
Comment (TipzRickyCheung) [reply app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt]: Done

File: app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt (hunk 1 above)
Comment (TipzRickyCheung) [reply app/src/main/java/eu/darken/capod/main/ui/widget/WidgetProvider.kt:115]: I've added some comments explaining this segment of code. Hopefully it's enough.

**Review by d4rken — APPROVED** [review]
lgtm :+1:

**Pull request merged**